lib/ext_json.js

```diff
@@ -230,8 +230,10 @@ function serializeDocument(doc, options) {
     } else if (val != null && typeof val === 'object') {
       _doc[name] = serializeDocument(val, options);
     }
-
     _doc[name] = serializeValue(val, options);
+    if (val instanceof RegExp) {
+      _doc[name] = codecs['BSONRegExp'].toExtendedJSON({ pattern: val.source, options: val.flags });
```

Comment:
[`Regex.prototype.flags` is not supported in Node 4](http://node.green/#ES2015-built-in-extensions-RegExp-prototype-properties-RegExp-prototype-flags). You can use [this polyfill instead](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/flags#Polyfill).